src/core/packages/saved-objects/base-server-internal/src/constants.ts

```diff
@@ -241,3 +241,50 @@ export const HASH_TO_VERSION_MAP = {
   'visualization|4891c012863513388881fc109fec4809': '10.0.0',
   'workplace_search_telemetry|3d1b76c39bfb2cc8296b024d73854724': '10.0.0',
 };
+
+/**
+ * Types that are no longer registered and need to be removed
+ * As of 8.8, no new types are allowed to be removed.
+ * Removing saved object types is not backward compatible
+ */
+export const REMOVED_TYPES: string[] = [
+  'apm-services-telemetry',
+  'background-session',
+  'cases-sub-case',
+  'file-upload-telemetry',
+  // https://github.com/elastic/kibana/issues/91869
+  'fleet-agent-events',
+  // https://github.com/elastic/obs-dc-team/issues/334
+  'fleet-agents',
+  'fleet-agent-actions',
+  'fleet-enrollment-api-keys',
+  // replaced by guided-onboarding-guide-state in 8.6
+  'guided-setup-state',
+  // Was removed in 7.12
+  'ml-telemetry',
+  'server',
+  // https://github.com/elastic/kibana/issues/95617
+  'tsvb-validation-telemetry',
+  // replaced by osquery-manager-usage-metric
+  'osquery-usage-metric',
+  // Was removed in 8.1 https://github.com/elastic/kibana/issues/91265
+  'siem-detection-engine-rule-status',
+  // Was removed in 8.7 https://github.com/elastic/kibana/issues/130966
+  'siem-detection-engine-rule-execution-info',
+  // Was removed in 7.16
+  'timelion-sheet',
+  // Removed in 8.3 https://github.com/elastic/kibana/issues/127745
+  'ui-counter',
+  // Deprecated, no longer used since 7.13 https://github.com/elastic/kibana/pull/94923/files
+  'application_usage_transactional',
+  // Removed in 7.8.1 / 7.9.0 https://github.com/elastic/kibana/pull/69871
+  'maps-telemetry',
+  // Deprecated, no longer used since 8.7 https://github.com/elastic/kibana/pull/148530
+  'csp_rule',
+  // Removed in 8.8 https://github.com/elastic/kibana/pull/151116
+  'upgrade-assistant-telemetry',
+  // Removed in 8.8 https://github.com/elastic/kibana/pull/155204
+  'endpoint:user-artifact',
+  // Removed in 9.1 https://github.com/elastic/kibana/pull/216342
+  'investigation',
+].sort();
```

Comment:
I saw you moved it to this package to have it closer to the SavedObjectTypeRegistry, but as per my [other comment](https://github.com/elastic/kibana/pull/218139/files#r2053626046), we can probably place this on `core-saved-objects-server-internal`.

Reply:
And if we move it, we'll have to update [the documentation](https://github.com/elastic/kibana/blob/main/src/core/server/docs/kib_core_reviewing_so_type_pr.mdx?plain=1#L54) that refers to it.